be/src/exprs/cast_expr.cpp

```diff
@@ -1127,6 +1134,43 @@ class VectorizedCastExpr final : public Expr {
         }
         return result_column;
     };
+
+    bool is_compilable() const override {
+        return !AllowThrowException && IRHelper::support_jit(FromType) && IRHelper::support_jit(ToType);
+    }
+
+    StatusOr<LLVMDatum> generate_ir_impl(ExprContext* context, const llvm::Module& module, llvm::IRBuilder<>& b,
+                                         const std::vector<LLVMDatum>& datums) const override {
+        auto* l = datums[0].value;
+
+        if constexpr (FromType == TYPE_JSON || ToType == TYPE_JSON) {
+            return Status::NotSupported("JIT casting does not support JSON");
+        } else if constexpr (lt_is_decimal<FromType> || lt_is_decimal<ToType>) {
+            return Status::NotSupported("JIT casting does not support decimal");
+        } else {
+            LLVMDatum datum(b);
+            ASSIGN_OR_RETURN(datum.value, IRHelper::cast_to_type(b, l, FromType, ToType));
+
+            if constexpr ((lt_is_integer<FromType> || lt_is_float<FromType>)&&(lt_is_integer<ToType> ||
+                                                                               lt_is_float<ToType>)) {
+                if constexpr (std::numeric_limits<RunTimeCppType<ToType>>::max() <
+                              std::numeric_limits<RunTimeCppType<FromType>>::max()) {
+                    // Check overflow.
+                    ASSIGN_OR_RETURN(auto cast_l, IRHelper::cast_to_type(b, datum.value, ToType, FromType));
+                    llvm::Value* is_overflow = nullptr;
+                    if constexpr (lt_is_integer<FromType>) {
+                        is_overflow = b.CreateICmpNE(l, cast_l);
+                    } else if constexpr (lt_is_float<FromType>) {
+                        is_overflow = b.CreateFCmpONE(l, cast_l);
+                    }
```

Comment:
what about downward overflowing?

Reply:
Is there an issue with this 'if' check? `if constexpr ((std::numeric_limits<RunTimeCppType<ToType>>::max() <`

Reply:
Done.